src/text.c

```diff
@@ -361,7 +362,30 @@ SDB_API bool sdb_text_load_buf(Sdb *s, char *buf, size_t sz) {
 	while (ctx.pos < ctx.bufsz) {
 		load_process_single_char (&ctx);
 	}
-	load_flush_line (&ctx);
+	if (ctx.line_begin < ctx.bufsz) {
```

Comment:
If a file ends with a newline the last line with key/value pair will be processed twice  won't it?

Reply:
Last call of `load_process_single_char()` changes the state to `STATE_NEWLINE` but don't change `line_begin`

Reply:
> If a file ends with a newline the last line with key/value pair will be processed twice won't it?

Not exactly. Since the state is `STATE_NEWLINE` then, `load_flush_line()` will basically do nothing. But the strdup including the previous line would unnecessarily happen, which we can easily optimize this by checking for `STATE_NEWLINE` earlier. Good catch!